fff

```diff
@@ -368,6 +369,49 @@ trash() {
     fi
 }
 
+bulk_rename() {
+  local tmp_file="$(mktemp)"
+
+  # Write all input file names to temporary file and open it in an editor.
+  printf "%s\n" "${@:1:$#-1}" > "$tmp_file"
+  "${EDITOR:-vi}" "$tmp_file"
+
+  local commands_file="$(mktemp)"
+
+  # Add instructions to the file with commands.
+  echo '# This file will be executed when you close the editor.' > "$commands_file"
```

Comment:
Something like this would be better.

```sh
printf '%s\n%s\n' \
    "# This file will be executed when you close the editor." \
    "# This file will be executed when you close the editor."
```